File: Samples/android/Kotlin/Android-SDK-Demo-App/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/AdExperienceFragment.java
package com.vungle.samples.samplejava.ui.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.core.content.ContextCompat;
import androidx.fragment.app.Fragment;

import com.vungle.samples.samplejava.R;
import com.vungle.samples.samplejava.databinding.FragmentAdexperienceBinding;

public class AdExperienceFragment extends Fragment {

    protected String placementId;

    protected FragmentAdexperienceBinding _binding = null;

    public AdExperienceFragment() {
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        _binding = FragmentAdexperienceBinding.inflate(inflater, container, false);
        if (getArguments() != null) {
            placementId = getArguments().getString("placement_id");
        }
        return _binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        _binding.btnLoadAd.setOnClickListener(view1 -> {
            AdExperienceFragment.this.resetCallbackLabelColor();
            AdExperienceFragment.this.loadAd();
        });
        _binding.btnPlayAd.setOnClickListener(view12 -> AdExperienceFragment.this.playAd());
        _binding.btnDestroyAd.setOnClickListener(view13 -> AdExperienceFragment.this.destroyAd());
        _binding.lbPlacementId.setText(placementId);
    }

    protected void destroyAd() {
    }

    protected void playAd() {

    }

    protected void loadAd() {

    }

    private void resetCallbackLabelColor() {
        _binding.adContainer.removeAllViews();
        setTextColor(_binding.lbAdLoaded, R.color.lightGray);
        setTextColor(_binding.lbAdStart, R.color.lightGray);
        setTextColor(_binding.lbAdImpression, R.color.lightGray);
        setTextColor(_binding.lbAdClick, R.color.lightGray);
        setTextColor(_binding.lbOnAdLeftApplication, R.color.lightGray);
        setTextColor(_binding.lbOnRewarded, R.color.lightGray);
        setTextColor(_binding.lbAdEnd, R.color.lightGray);
        setTextColor(_binding.lbError, R.color.lightGray);

        _binding.lbError.setText(R.string.error);
    }

    protected void setTextColor(TextView textView, int color) {
        textView.setTextColor(ContextCompat.getColor(requireActivity(), color));
    }
}

File: Samples/android/Kotlin/app-v7-kotlin/src/main/java/com/vungle/samples/samplekotlin/ui/AdExperienceFragment.kt
package com.vungle.samples.samplekotlin.ui

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.activity.OnBackPressedCallback
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import com.vungle.samples.samplekotlin.utils.extensions.blackText
import com.vungle.samples.samplekotlin.utils.extensions.lightGrayText
import com.vungle.ads.BaseAd
import com.vungle.ads.InterstitialAdListener
import com.vungle.ads.RewardedAdListener
import com.vungle.ads.VungleError
import com.vungle.samples.samplekotlin.R
import com.vungle.samples.samplekotlin.databinding.FragmentAdexperienceBinding

open class AdExperienceFragment : Fragment(R.layout.fragment_adexperience), InterstitialAdListener, RewardedAdListener {

  protected open val TAG = "VungleAds"

  private var _binding: FragmentAdexperienceBinding? = null
  val binding get() = _binding!!
  lateinit var placementId: String

  override fun onCreateView(
    inflater: LayoutInflater,
    container: ViewGroup?,
    savedInstanceState: Bundle?
  ): View? {
    _binding = FragmentAdexperienceBinding.inflate(inflater, container, false)
    placementId = arguments?.getString("placement_id").toString()
    return binding.root
  }

  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
      binding.nativeVideoLog.visibility = View.GONE
    binding.btnLoadAd.setOnClickListener {
      resetCallbackLabelColor()
      loadAd()
    }
    binding.btnPlayAd.setOnClickListener {
      playAd()
    }
    binding.btnDestroyAd.setOnClickListener {
      destroyAd()
    }
    requireActivity().onBackPressedDispatcher.addCallback(
      viewLifecycleOwner,
      object : OnBackPressedCallback(
        true
      ) {
        override fun handleOnBackPressed() {
          findNavController().navigateUp()
        }
      })
  }

  override fun onAdLoaded(baseAd: BaseAd) {
    Log.d(TAG, "adLoaded")
    binding.lbAdLoaded.blackText()
    binding.btnPlayAd.isEnabled = true
  }

  override fun onAdStart(baseAd: BaseAd) {
    Log.d(TAG, "adStart")
    binding.lbAdStart.blackText()
    binding.btnPlayAd.isEnabled = false
  }

  override fun onAdImpression(baseAd: BaseAd) {
    Log.d(TAG, "adImpression")
    binding.lbAdImpression.blackText()
  }

  override fun onAdEnd(baseAd: BaseAd) {
    Log.d(TAG, "adEnd")
    binding.lbAdEnd.blackText()
  }

  override fun onAdClicked(baseAd: BaseAd) {
    Log.d(TAG, "adClick")
    binding.lbAdClick.blackText()
  }

  override fun onAdRewarded(baseAd: BaseAd) {
    Log.d(TAG, "adRewarded")
    binding.lbOnRewarded.blackText()
  }

  override fun onAdLeftApplication(baseAd: BaseAd) {
    Log.d(TAG, "onAdLeftApplication")
    binding.lbOnAdLeftApplication.blackText()
  }

  override fun onAdFailedToLoad(baseAd: BaseAd, adError: VungleError) {
    val exceptionCode = adError.code
    val localizedMessage = adError.localizedMessage
    val errorMessage = "onAdFailedToLoad - $exceptionCode $localizedMessage"
    Log.d(TAG, errorMessage)
    binding.lbError.apply {
      blackText()
      text = errorMessage
    }
  }

  override fun onAdFailedToPlay(baseAd: BaseAd, adError: VungleError) {
    val exceptionCode = adError.code
    val localizedMessage = adError.localizedMessage
    val errorMessage = "onAdFailedToPlay - $exceptionCode $localizedMessage"
    Log.d(TAG, errorMessage)
    binding.lbError.apply {
      blackText()
      text = errorMessage
    }
  }

  protected open fun loadAd() {
    Log.d(TAG, "Load ad")
  }

  protected open fun playAd() {
    Log.d(TAG, "Play ad")
  }

  protected open fun destroyAd() {
    Log.d(TAG, "Destroy ad")
  }

  open fun resetCallbackLabelColor() {
    binding.adContainer.removeAllViews()
    binding.lbAdLoaded.lightGrayText()
    binding.lbAdStart.lightGrayText()
    binding.lbAdImpression.lightGrayText()
    binding.lbAdClick.lightGrayText()
    binding.lbOnAdLeftApplication.lightGrayText()
    binding.lbOnRewarded.lightGrayText()
    binding.lbAdEnd.lightGrayText()
    binding.lbError.lightGrayText()
    binding.lbError.setText(R.string.error)
  }

  override fun onDestroyView() {
    super.onDestroyView()
    _binding = null
  }
}